src/update.c

```diff
@@ -315,25 +315,28 @@ static int main_update()
 	grabtime_stop(&times);
 	grabtime_stop(&times); // Close step 1
 	grabtime_start(&times, "Load Manifests:");
+	int manifest_err;
 load_current_mom:
 	/* Step 3: setup manifests */
 
 	/* get the from/to MoM manifests */
 	if (system_on_mix()) {
-		current_manifest = load_mom(current_version, false, mix_exists);
+		current_manifest = load_mom(current_version, false, mix_exists, &manifest_err);
 	} else {
-		current_manifest = load_mom(current_version, false, false);
+		current_manifest = load_mom(current_version, false, false, &manifest_err);
 	}
 	if (!current_manifest) {
 		/* TODO: possibly remove this as not getting a "from" manifest is not fatal
 		 * - we just don't apply deltas */
-		if (retries < MAX_TRIES) {
+		if (retries < MAX_TRIES && manifest_err != -ENET404 && manifest_err != -ENOENT) {
 			increment_retries(&retries, &timeout);
-			fprintf(stderr, "Retry #%d downloading from/to MoM Manifests\n", retries);
+			fprintf(stderr, "Retry #%d downloading current MoM manifest\n", retries);
 			goto load_current_mom;
 		}
-		fprintf(stderr, "Failure retrieving manifest from server\n");
-		ret = EMOM_NOTFOUND;
+		if (manifest_err == -ENET404 || manifest_err == -ENOENT) {
+			fprintf(stderr, "The current MoM manifest was not found\n");
+		}
```

Comment:
I think this would be easier to read:

if (manifest_err == -ENET404 || manifest_err == -ENOENT) {
....
} else if (retries < MAX_TRIES) {
.....
}

Reply:
ok, I'll make that change @jwakre